### PR TITLE
Fix conn_graph_facts issues of `anchor` and `filename` options

### PR DESCRIPTION
**Pull request by wangxin**
### Description of PR

Summary:
Fixes # (issue)

Fix the conn_graph_facts issues introduced in recent changes.

### Type of change

- [x] Bug fix
- [ ] Testbed and Framework(new/improvement)
- [ ] Test case(new/improvement)

#### What is the motivation for this PR?
The recent changes for conn_graph_facts introduced two issues:

* `anchor` option issue
When use `anchor` to get the whole connection graph facts for configuring root
fanout switch, the `host` and `hosts` options are not given. So the `hostnames`
variable will have value [None] which will not be evaluated to False. Then the
`target` variable will take the value of `hostnames`, i.e. [None]. Eventually the empty
graph file will be used.

* `filename` option issue
When the `filename` argument is specified, need to open the specified file under
path LAB_GRAPHFILE_PATH.

#### How did you do it?
* Fix the issue of parsing wrong connection graph file when the `anchor` option is used.
* Fix the issue of unable to find connection graph file when the `filename` option is used.
* Set host, hosts and anchor mutually exclusive.
* Update the documentation.

#### How did you verify/test it?
Prepared a temporary ansible playbook to call the conn_graph_facts module with different option combinations, including negative combinations.

### CODE REVIEW

**Review by lolyu — APPROVED** [review]
LGTM

File: ansible/library/conn_graph_facts.py [hunk 1]
```diff
@@ -17,8 +17,28 @@
     Retrive lab fanout switches physical and vlan connections
     add to Ansible facts
 options:
-    host:  [fanout switch name|Server name|Sonic Switch Name]
-    requred: True
+    host:
+        [fanout switch name|Server name|Sonic Switch Name]
+        requred: False
```
Comment (neethajohn) [inline ansible/library/conn_graph_facts.py:22]: spell check 'required'

File: ansible/library/conn_graph_facts.py [hunk 2]
```diff
@@ -291,14 +311,14 @@ def main():
             LAB_GRAPHFILE_PATH = m_args['filepath']
 
         if m_args['filename']:
-            filename = m_args['filename']
+            filename = os.path.join(LAB_GRAPHFILE_PATH, m_args['filename'])
```
Comment (neethajohn) [inline ansible/library/conn_graph_facts.py:314]: Why do we have a requirement that the file has to be in this path?

File: ansible/library/conn_graph_facts.py (hunk 1 above)
Comment (wangxin) [reply ansible/library/conn_graph_facts.py]: Thanks! Fixed.

File: ansible/library/conn_graph_facts.py (hunk 2 above)
Comment (wangxin) [reply ansible/library/conn_graph_facts.py:314]: Without the path, ansible will try to lookup the graph files under the CWD. We need to tell ansible where to lookup the graph files. Generally, all the graph files are stored under `LAB_GRAPHFILE_PATH`. This global variable has default value `sonic-mgmt/ansible/files`. Its value can be overridden by passing in argument `filepath` while calling the `conn_graph_facts` module. No matter use default value or use customized value, `LAB_GRAPHFILE_PATH` is always for location of the graph files. So, we should always try to lookup graph files under `LAB_GRAPHFILE_PATH `

**Review by neethajohn — APPROVED** [review]

**Comment by wangxin** [comment]
retest this please

**Pull request merged**